app/assets/javascripts/admin/modules/breadcrumb_preview.js

```diff
@@ -0,0 +1,66 @@
+(function(Modules) {
+  "use strict";
+
+  Modules.BreadcrumbPreview = function() {
+    var that = this;
+
+    that.fetchCurrentCheckboxes = function() {
+      return $('.topic-tree :checked');
+    };
+
+    that.buildBreadcrumbsStructure = function(checkboxes) {
+      var structure = $.map(checkboxes, function(checkbox) {
+        var $checkbox = $(checkbox);
+        var ancestors = $checkbox.data('ancestors').split('|');
+
+        return [ancestors];
+      });
+
+      return structure;
+    };
+
+    that.filterBreadcrumbs = function(breadcrumbs) {
+      var longestFirst = breadcrumbs.sort(function (a, b) {
+        return b.length - a.length;
+      });
+
+      var seenAlready = [];
+
+      $.each(longestFirst, function(index, breadcrumb) {
+        var seen = false;
+
+        $.each(seenAlready, function(index, seenBreadcrumb) {
+          var breadcrumbString = JSON.stringify(breadcrumb);
+          var seenBreadcrumbString = JSON.stringify(seenBreadcrumb.slice(0, breadcrumb.length));
+
+          if (breadcrumbString === seenBreadcrumbString) {
+            seen = true;
+          }
+        });
+
+        if (!seen) {
+          seenAlready.push(breadcrumb);
+        }
+      });
+
+      return seenAlready;
+    }
+
+    that.start = function(element) {
+      var $topicTree = $('.topic-tree');
+
+      $topicTree.on('change', function () {
```

Comment:
Since this is not [debounced](https://remysharp.com/2010/07/21/throttling-function-calls), I'd worry there are performance problems with this approach.

Have you profiled this on a slower computer? You could try CPU Throttling to x10+ (https://umaar.com/dev-tips/88-cpu-throttling/)